src/printer.ts

```diff
@@ -194,6 +199,48 @@ export class PugPrinter {
 						results.push(this[token.type](token));
 						break;
 				}
+				if (
+					token.type === 'comment' &&
+					token.val.trim() === 'prettier-ignore'
+				) {
+					// Use a separate token processing loop that finds the end of the tokens to be ignored by formatting,
+					// and uses their `loc` properties to retrieve the original pug code to be used instead.
+					const firstToken: Token = this.tokens[index];
+					const startsWithNewLine: boolean = firstToken.type === 'newline';
+					let skipNewline: boolean = startsWithNewLine;
+					let ignoreLevel: number = 0;
+					while (index < this.tokens.length) {
```

Comment:
This makes me curious :thinking:  
You are accessing and also **mutate** the same `index` as in the outer while loop?

Does this always work? Or did you just catch special cases with your current tests?

Reply:
Yes this works, and it's the reason why I switched to `while` loops from `for`. The looks themselves just check the condition, they don't increase the index. You can have an inner loop that increases the same index as the outer loop. The inner loop usually stops because of one of the `break` statements inside of it. The `while` condition is just there to make sure it never loops infinitely.  You could replace it with `while(true) {` and it would still work.

Reply:
I will try to add a `getNextToken()` helper that internally increases `this.currentIndex` so that the whole thing becomes a bit more like reading from a stream. And the loop and nested loop can then both iterate until this helper returns `null`.

Does this sound better?

Reply:
I've managed to implement it exactly like this, and it works well I think!